tests/D2L.CodeStyle.Analyzers.Test/UnsafeStatics/UnsafeStaticsAnalyzerTests.cs

```diff
@@ -672,6 +674,61 @@ internal class Bar {
 			AssertSingleDiagnostic( test, 7, 41, "foo", "test.Tests.Foo" );
 		}
 
+		[Test]
+		public void DocumentWithStaticField_ReadonlyUnsafeBaseClassWithUnsafeInitializer_Diag() {
+			const string test = @"
+	using System;
+	namespace test {
+		class Tests {
+			interface IUnsafe { void Magic(); } // could be anythinggggggg
+
+			class Safe : IUnsafe {
+				void IUnsafe.Magic() {} // looks safe to me
+				public static readonly Safe Instance { get; } = new Safe();
+			}
+
+			private readonly static IUnsafe foo = Safe.Instance;
+		}
+	}";
+
+			// Safe isn't sealed
+			AssertSingleDiagnostic( test, 12, 36, "foo", "test.Tests.Safe" );

```

Comment:
Well all static fields need to be audited anyway, so they're either (1) deterministically safe, (2) annotated with `audited`, or (3) clear/unaudited. In the former two cases, we should be able to come to the same conclusion one level upstream (gets tricky if we don't have the source code, which we often won't). I've got some thoughts on how to do that in my head, will take a crack after the immutable analyzer.